tests/utils.py

```diff
@@ -223,7 +226,7 @@ def __init__(self, bitcoin_dir="/tmp/bitcoind-test", rpcport=18332):
 
     def start(self):
         TailableProc.start(self)
-        self.wait_for_log("Done loading", timeout=10)
+        self.wait_for_log("Done loading", timeout=None)
```

Comment:
Not sure about this. Even though it "should" always hit this state, I don't think it's a good idea to rely on this always being true. Is there any compelling reason to remove this timeout, or any timeout for that matter?

Reply:
In practice, this timeout caused an issue for me (and possibly other people), because on my system it took bitcoind 11..12 seconds to produce "Done loading". The lazy solution would be to just increase the timeout to something that "works for me" and that is so ridiculously large that I expect it to reliably work for all. What would be a good value?

I think it's a trade-off. With a too low value, some people (like me) get failures that have nothing to do with bugs in lightningd, or worse, get inconsistent results (sometimes success, sometimes not). With a too high value, it takes a long time for people to discover that something is wrong with bitcoind. That case should be consistent though (assuming bitcoind is consistent), making it easier to debug. Also, it should be a more unlikely case: I see bitcoind as part of the test environment, not part of the system under test. So we assume bitcoind to be bug-free, and only test behavior of lightningd. Assuming a perfect bitcoind, it should either succeed, or crash (e.g. if the OS doesn't allow it to do something essential).

All in all, I see the second case (unlikely, easy to debug) as better than the first case (possible, potentially hard to debug). The trade-off leads me to choose an infinite time-out.

Reply:
I strongly disagree: having a test hang forever is definitely worse than having an indication that the test failed due to a timeout. We could talk about having a modifier for timeouts, e.g., be able to say via an environment variable that timeouts should be twice as long on this system, but disabling them altogether is guaranteed to cause us more issues down the line.

Reply:
I gave it some more thought, and realized that it might be inconvenient for a person running a test that there is no way to distinguish between a hanging test and a test that just takes a long time. I do agree there should be time-outs in the code handling the lightningd process, since the test code can not a-priori assume sane behavior of lightningd.

For bitcoind, which can be assumed to be sane and non-hanging(*), there is no strong need for a time-out. You could include one as a precaution, but it has the disadvantage of causing issues on systems that are slower than usual. If you do insist on having a time-out here, it should be *insanely* large to eliminate that problem, or, in other words, an order of magnitude larger. Since I measured 11 or 12 seconds, the time-out should certainly not be lower than 60 seconds.

(*)if any counter-examples exist, they are likely things that should be fixed in bitcoind.